_modules/qubes_dom0_update.py

```diff
@@ -3311,3 +3319,109 @@ def list_installed_patches():
         salt '*' pkg.list_installed_patches
     '''
     return _get_patches(installed_only=True)
+
+def mod_init(low):
+    '''
+    Set a flag to tell the install functions to refresh the package database.
+    This ensures that the package database is refreshed only once during
+    a state run significantly improving the speed of package management
+    during a state run.
+
+    It sets a flag for a number of reasons, primarily due to timeline logic.
+    When originally setting up the mod_init for pkg a number of corner cases
+    arose with different package managers and how they refresh package data.
+
+    It also runs the "ex_mod_init" from the package manager module that is
+    currently loaded. The "ex_mod_init" is expected to work as a normal
+    "mod_init" function.
+
+    .. seealso::
+       :py:func:`salt.modules.ebuild.ex_mod_init`
+
+    '''
+    ret = True
+    if 'pkg.ex_mod_init' in __salt__:
+        ret = __salt__['pkg.ex_mod_init'](low)
+
+    # QUBES-DOM0: refresh packages on 'uptodate', too
+    if low['fun'] in ('installed', 'latest', 'uptodate'):
+        salt.utils.pkg.write_rtag(__opts__)
+        return ret
+    return False
+
+
+def mod_aggregate(low, chunks, running):
```

Comment:
This, and the `mod_watch` I think are not really needed there. They are done at the `salt.states.pkg` level (this file provides `salt.modules.pkg`).